raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -739,16 +739,20 @@ def events_for_expired_locks(
         assert channel_state, "Couldn't find channel for channel_id: {}".format(channel_identifier)
 
         secrethash = mediator_state.secrethash
-        locked_lock = channel_state.our_state.secrethashes_to_lockedlocks.get(secrethash)
+        lock = None
+        if secrethash in channel_state.our_state.secrethashes_to_lockedlocks:
```

Comment:
I would also add an assertion here. That if it's in `secrethashes_to_lockedlocks` it's not in `secrethashes_to_unlockedlocks` and vice versa.

Reply:
@rakanalh can you please fix this function?

- It's not taking reorgs into account (it's not waiting for confirmation blocks to make sure the secret was really not registered)
- It's not checking if the lock was unlocked on-chain

To fix this I would remove this code:

```python
            unlock_failed = EventUnlockFailed(
                pair.payee_transfer.payment_identifier,
                pair.payee_transfer.lock.secrethash,
                'lock expired',
            )
            events.append(unlock_failed)
```

To the `events_to_remove_expired_locks` function, and leave the `EventUnlockClaimFailed` in this function, however, doing the expiration check using `channel.is_lock_expired` /with/ the confirmation blocks.

Reply:
@LefterisJP 
> I would also add an assertion here. That if it's in secrethashes_to_lockedlocks it's not in secrethashes_to_unlockedlocks and vice versa.

Might be a better idea to assert this in the tests, no? because: 
1. We shouldn't add more asserts IMO
2. Whenever the lock is unlocked, it is either moved to unlocked_locks or onchain_unlocked_locks

Reply:
@rakanalh 

> Whenever the lock is unlocked, it is either moved to unlocked_locks or onchain_unlocked_locks

Yeah that's what I would like the assert to check. Assertions are generally good to check invariants and fail early. 

It is true that we could have those assertions in the test but having them in the code also documents our assumptions about the code, makes it clear to anyone who reads what the invariants are and if anything changes things will break, fail early and let us know exactly why.

Reply:
If you have a strong opinion about having the asserts to be explicit, i could add them... but i believe the if statement there implicitly does this:
```

Reply:
@rakanalh sorry, I'm talking about a different function, the function [events_for_expired_pairs](https://github.com/raiden-network/raiden/blob/38971b372dafb3205cbd3df8cfc3306922a55eac/raiden/transfer/mediated_transfer/mediator.py#L542) is not using `channel.is_lock_expired`

Reply:
@hackaugusto implemented as requested.

Reply:
@rakanalh I don't understand how you think implicit assertion works in the first if? What stops the lock also being in `unlocked_locks` even if we get into the first if?

Reply:
Because the state of the lock goes only forward and not backward. Meaning that we add the secrethash to lockedlocks and then move it forward to unlockedlocks or onchain_unlockedlocks but not the other way around. But anyway i added the asserts to satisfy your request.

Reply:
yeah exactly this is what the assert is for. To make sure it's going forward and not backward. This is the invariant you are checking.